File: ios/FastCreateHash.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(FastCreateHash, NSObject)

RCT_EXTERN_METHOD(multiply:(float)a
                  withB:(float)b
                  withResolver:(RCTPromiseResolveBlock)resolve
                  withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(createHash:(NSString *)data64
                  withAlgorithm:(NSString *)algorithm
                  withResolver:(RCTPromiseResolveBlock)resolve
                  withRejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/FastCreateHash.swift
import Foundation

@objc(FastCreateHash)
class FastCreateHash: NSObject {
    // https://stackoverflow.com/a/50294485
    @objc static func requiresMainQueueSetup() -> Bool {
        return false
    }

    @objc(multiply:withB:withResolver:withRejecter:)
    func multiply(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        resolve(a*b)
    }
}
